### ISSUE TITLE
EGL crashes frequently

### ISSUE THREAD

**Issue by wookayin** [deepmind/lab#134]
I am using `egl` as hardware backend for headless rendering.

When resetting the environment via `env.reset()` or during running an environment through normal interactions, I can quite frequently observe the following error:

```
EGL ERROR: 0x3001 file:engine/code/deepmind/headless_egl_glimp.c, line:67
```

Note that the 0x3001 error is `EGL_NOT_INITIALIZED.` Why does it happen? Looked similar to #62, but I was unable to find any clue.

I attach [crashlog.txt](https://github.com/deepmind/lab/files/2872599/crashlog.txt) for diagnosis of the problem.

**Comment by tkoeppe** [comment]
Is this related to starting multiple environments in the same process at once?

**Comment by wookayin** [comment]
Sorry for not providing enough details for the issue. Yes, I was creating multiple Lab instances (though only a few, like 2-3 even has the same issue) in the same process (not in parallel, just from the main thread).

Specifically I can see this happening frequently when I run a code under *jupyter notebooks*. Should I just create only one `Lab` environment per process?

**Comment by alex-petrenko** [comment]
Unfortunately, this happens to me from time to time as well.

I also found a 100% way to reproduce this error (0x3001):
1) create two Lab environments in the same process, I use Python API
2) call reset() and then step() on each at least once
3) call close() on each environment, the second call is guaranteed to crash with the same EGL error

**Comment by tkoeppe** [comment]
@alex-petrenko: is this explained by #62?

**Comment by alex-petrenko** [comment]
@tkoeppe yes, I guess, thanks for the link.
What I found out is that most of the time running multiple DMLabs in the same process actually works okay, and if it crashes (with 3001 or 3002) it only happens in the very beginning or the very end. So for my purposes, I'm still using EGL renderer, because it's just so much faster than software renderer. I am currently running 96 environments across 16 processes (for RL).

That said, I was able to run much more "software" environments, there does not seem to be any limit, other than the amount of RAM that I have, I was able to run 256 environments on a single PC.
